File: catkin_ws/src/learning_service/scripts/person_server.py
#!/usr/bin/env python
import rospy
from learning_service.srv import Person, PersonResponse

def personCallback(req):
    # show request
    rospy.loginfo("Person: name:%s  age:%d  sex:%d", req.name, req.age, req.sex)

    # return response
    return PersonResponse("OK")

def person_server():
    # ROS node init
    rospy.init_node("person_server")

    # create server /person_server, callback function is personCallback
    s = rospy.Service('/show_person', Person, personCallback)

    # loop wait
    print "Ready to receive person information."
    rospy.spin()

if __name__ == '__main__':
    person_server()

File: catkin_ws/src/learning_service/src/person_server.cpp
/**
 * execute /show_person service, service data type is learning_service::Person
 */

#include <ros/ros.h>
#include "learning_service/Person.h"

// service callback function
bool personCallback(learning_service::Person::Request &req, learning_service::Person::Response &res)
{
	// show request
	ROS_INFO("Person: name:%s  age:%d  sex:%d", req.name.c_str(), req.age, req.sex);

	// set response
	res.result = "OK";

	return true;
}

int main(int argc, char **argv)
{
	// ROS node init
	ros::init(argc, argv, "person_server");

	// create handler
	ros::NodeHandle n;

	// create service /show_person, callback function is personCallback
	ros::ServiceServer person_server = n.advertiseService("/show_person", personCallback);

	ROS_INFO("Ready to receive person information.");
	ros::spin();
	
	return 0;
}
